profiling/src/allocation/mod.rs

```diff
@@ -84,11 +87,24 @@ pub fn collect_allocation(len: size_t) {
 }
 
 thread_local! {
-    static ALLOCATION_PROFILING_STATS: RefCell<AllocationProfilingStats> =
-        RefCell::new(AllocationProfilingStats::new());
+    pub(crate) static ALLOCATION_PROFILING_STATS: MaybeUninit<UnsafeCell<AllocationProfilingStats>> =
```

Comment:
I was wrong before, this should be an `UnsafeCell<MaybeUninit<T>>` and not `MaybeUninit<UnsafeCell<T>>`. The reason is your initialization code below is undefined behavior (I'll leave a comment there too), from [MaybeUninit::as_ptr](https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#method.as_ptr):

> Writing to memory that this pointer (non-transitively) points to is undefined behavior (except inside an `UnsafeCell<T>`).

But we have the opposite. We have to invert.